Demote x86_64-apple-darwin to Tier 2 with host tools

shepmaster opened the pull request:
Switch to only using aarch64 runners (implying we are now cross-compiling) and stop running tests. In the future, we could enable (some?) tests via Rosetta 2.

This implements the decision from https://github.com/rust-lang/rfcs/pull/3841.

shepmaster committed:
Demote x86_64-apple-darwin to Tier 2 with host tools

Switch to only using aarch64 runners (implying we are now
cross-compiling) and stop running tests. In the future, we could
enable (some?) tests via Rosetta 2.

shepmaster commented:
@bors2 try jobs=dist-x86_64-apple

rust-log-analyzer commented:
The job **`dist-x86_64-apple`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/16887597483/job/47839174722) [(plain enhanced)](https://triage.rust-lang.org/gha-logs/rust-lang/rust/47839174722) [(plain)](https://github.com/rust-lang/rust/commit/c20c6a6c5d536319b8048b888c8e9b20c9226ab1/checks/47839174722/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
pkgbuild: Adding top-level postinstall script
pkgbuild: Wrote package to /Users/runner/work/rust/rust/build/tmp/dist/combined-tarball/pkg/miri.pkg

thread 'main' panicked at src/bootstrap/src/lib.rs:1934:25:
could not read dir "/Users/runner/work/rust/rust/build/tmp/tarball/rust/x86_64-apple-darwin/rustc-codegen-cranelift-nightly-x86_64-apple-darwin": Os { code: 2, kind: NotFound, message: "No such file or directory" }
stack backtrace:
   0: __rustc::rust_begin_unwind
             at /rustc/788da80fcfcef3f34c90def5baa32813e39a1a41/library/std/src/panicking.rs:697:5
   1: core::panicking::panic_fmt
             at /rustc/788da80fcfcef3f34c90def5baa32813e39a1a41/library/core/src/panicking.rs:75:14
```

</details>

shepmaster committed:
Use aarch64-apple-darwin as the fallback doc source for `-apple-`

We are moving away from `x86_64-apple-darwin`, so soon these docs
won't be available.

shepmaster commented:
@bors2 try jobs=dist-x86_64-apple

madsmtm reviewed (approved):
Compiler parts look good to me, thanks! Feel free to r=me once you figure out the CI

rust-log-analyzer commented:
The job **`dist-x86_64-apple`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/16993248238/job/48177662633) [(plain enhanced)](https://triage.rust-lang.org/gha-logs/rust-lang/rust/48177662633) [(plain)](https://github.com/rust-lang/rust/commit/63227c320ddccb28106c6afb5067bb8615d36318/checks/48177662633/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
   Compiling opener v0.5.2
   Compiling termcolor v1.4.1
error: linking with `cc` failed: exit status: 1
  |
  = note:  "cc" "/var/folders/y6/nj790rtn62lfktb1sh__79hc0000gn/T/rustcAs4FDv/symbols.o" "<61 object files omitted>" "/Users/runner/work/rust/rust/build/bootstrap/debug/deps/{libfd_lock-79362d4bc8e0abed,libbootstrap-94fe542c9bd379c8,libtar-ce2beea407f6eaaf,libxattr-cfcd48e57f011319,librustix-133683f8c2896bb3,liberrno-1de382d1b68feaa5,libfiletime-5c18d064213f9992,libsemver-0052f93ace7a7d67,libopener-1a854111fdf5f15d,libhome-48c341e8ec48ed69,libcmake-c1168e0bf837cddc,libserde_json-878af40c795563fd,libitoa-baeee1228b5c496d,libryu-203fda4bf3b69f89,libtoml-b166382e88ec84e1,libsysinfo-91cd0e7289c595ff,libobjc2_io_kit-7c893838bb5473e6,libobjc2_core_foundation-e1208cac626d7912,libbitflags-e17c68d233b63596,libxz2-666ee1da07f643f9,liblzma_sys-16f5b00f1687edba,liblibc-4dbde612441b73de,libsha2-b16d5f21941b0cc6,libcpufeatures-f3b30edd6535e49b,libcfg_if-648b03fa4d97f7b4,libdigest-ccde569046a567e4,libblock_buffer-b30e02f98d3f55f4,libcrypto_common-1284c492923525a5,libgeneric_array-ba7f962e8ad0c208,libtypenum-0c320bb620fef05d,libclap_complete-17682b9ea1695d47,libclap-84d7887c5f09677c,libclap_builder-07df7048596dce31,libclap_lex-f5186b05187bea9b,libanstyle-fba4e8bbdf2d14cf,libignore-25db016f9ec7b307,libwalkdir-d261981939b59828,libsame_file-2a779e4614bb1148,libcrossbeam_deque-3588c3048b04de02,libcrossbeam_epoch-1032f529ecd0cf01,libcrossbeam_utils-6470300e563160c7,libglobset-e36bdcb69143edff,liblog-2ef03d3d5556b096,libregex_automata-6634f752558f8cd0,libregex_syntax-b5bbf549091626b3,libbstr-21f0052b4270e866,libaho_corasick-fbba079ed49bc7f1,libobject-5e6c60c511864f09,libmemchr-cce746d5357f8154,libtermcolor-62cd56008319d87a,libcc-5bb9606bba0b3119,libshlex-486306361509cf3c,libbuild_helper-834b7cd8ffaa0d75,libserde-7502344651e4d58c}.rlib" "<sysroot>/lib/rustlib/x86_64-apple-darwin/lib/{libstd-*,libpanic_unwind-*,libobject-*,libmemchr-*,libaddr2line-*,libgimli-*,librustc_demangle-*,libstd_detect-*,libhashbrown-*,librustc_std_workspace_alloc-*,libminiz_oxide-*,libadler2-*,libunwind-*,libcfg_if-*,liblibc-*,librustc_std_workspace_core-*,liballoc-*,libcore-*,libcompiler_builtins-*}.rlib" "-framework" "IOKit" "-framework" "CoreFoundation" "-llzma" "-liconv" "-lSystem" "-lc" "-lm" "-arch" "x86_64" "-mmacosx-version-min=10.12.0" "-L" "/opt/homebrew/Cellar/xz/5.8.1/lib" "-o" "/Users/runner/work/rust/rust/build/bootstrap/debug/deps/bootstrap-1ee7abb8415a1697" "-Wl,-dead_strip" "-nodefaultlibs"
  = note: some arguments are omitted. use `--verbose` to show all linker arguments
  = note: ld: warning: ignoring file '/opt/homebrew/Cellar/xz/5.8.1/lib/liblzma.5.dylib': found architecture 'arm64', required architecture 'x86_64'
          ld: Undefined symbols:
            _lzma_code, referenced from:
                xz2::stream::Stream::process::h6ddd924cba15cf32 in libxz2-666ee1da07f643f9.rlib[3](xz2-666ee1da07f643f9.xz2.f801b2ac39ffaab5-cgu.0.rcgu.o)
            _lzma_end, referenced from:
                _$LT$xz2..stream..Stream$u20$as$u20$core..ops..drop..Drop$GT$::drop::h51155f283065a3a4 in libxz2-666ee1da07f643f9.rlib[3](xz2-666ee1da07f643f9.xz2.f801b2ac39ffaab5-cgu.0.rcgu.o)
            _lzma_stream_decoder, referenced from:
                xz2::stream::Stream::new_stream_decoder::h173cdd5b9f740e16 in libxz2-666ee1da07f643f9.rlib[3](xz2-666ee1da07f643f9.xz2.f801b2ac39ffaab5-cgu.0.rcgu.o)
          clang: error: linker command failed with exit code 1 (use -v to see invocation)
          

error: could not compile `bootstrap` (bin "bootstrap") due to 1 previous error
warning: build failed, waiting for other jobs to finish...
failed to run: /Users/runner/work/rust/rust/build/x86_64-apple-darwin/stage0/bin/cargo build --manifest-path /Users/runner/work/rust/rust/src/bootstrap/Cargo.toml -Zroot-dir=/Users/runner/work/rust/rust --locked --features build-metrics
Build completed unsuccessfully in 0:03:49
  local time: Fri Aug 15 15:35:27 UTC 2025
  network time: Fri, 15 Aug 2025 15:35:28 GMT
```

</details>

shepmaster commented:
@bors2 try jobs=dist-x86_64-apple

shepmaster commented:
@bors r=Kobzol,madsmtm rollup

bors commented:
:pushpin: Commit c574c91e5739519b2fd3fdb5f07fa6e102705703 has been approved by `Kobzol,madsmtm`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=Kobzol,madsmtm c574c91e5739519b2fd3fdb5f07fa6e102705703 -->
<!-- homu: {"type":"Approved","sha":"c574c91e5739519b2fd3fdb5f07fa6e102705703","approver":"Kobzol,madsmtm","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->

shepmaster commented:
For record keeping, @Kobzol [approved on Zulip](https://rust-lang.zulipchat.com/#narrow/channel/242791-t-infra/topic/macos-13.20shutting.20down/near/535004603)
